llvm-general/src/LLVM/General/Internal/FFI/PassManagerC.cpp

```diff
@@ -85,8 +87,16 @@ void LLVM_General_AddGlobalValueNumberingPass(LLVMPassManagerRef PM, LLVMBool no
 }
 
 void LLVM_General_AddInternalizePass(LLVMPassManagerRef PM, unsigned nExports, const char **exports) {
-	std::vector<const char *> exportList(exports, exports + nExports);
-	unwrap(PM)->add(createInternalizePass(exportList));
+    const std::vector<const char *> exportList(exports, exports + nExports);
```

Comment:
You'll have copied the array of character pointers, but not the strings the pointers point to. I think you need to use a vector<string> or similar so the lambda owns string data.